consensus/src/ledger/indexed_merkle_tree.rs

```diff
@@ -0,0 +1,129 @@
+// Copyright (C) 2019-2021 Aleo Systems Inc.
+// This file is part of the snarkOS library.
+
+// The snarkOS library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The snarkOS library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the snarkOS library. If not, see <https://www.gnu.org/licenses/>.
+
+use std::{marker::PhantomData, sync::Arc};
+
+use anyhow::*;
+use indexmap::IndexSet;
+use snarkos_storage::Digest;
+use snarkvm_algorithms::{merkle_tree::MerkleTree, MerkleParameters};
+use snarkvm_utilities::ToBytes;
+
+pub struct IndexedMerkleTree<P: MerkleParameters> {
+    tree: MerkleTree<P>,
+    index: IndexSet<Digest>,
```

Comment:
This should instead be `digests` or `indexed_digests`

Reply:
thats fine and i'll do that because why not, but the main idea is context-dependence makes it clear what's going on, and we can have a simpler name as a result.
same reason you generally dont have a `mymodule::MyModuleStuff` struct as opposed to a `mymodule::Stuff` struct.

Reply:
the `index` name is still there; it suggests that it's a single value as opposed to a set